xarray/plot/dataarray_plot.py

```diff
@@ -948,6 +948,12 @@ def newplotfunc(
             size_ = kwargs.pop("_size", linewidth)
             size_r = _LINEWIDTH_RANGE
 
+        # Remove any nulls, .where(m, drop=True) doesn't work when m is a dask array,
```

Comment:
This can only be done for scatter plots. A line plot would totally change.

Reply:
Currently only scatter uses `plot1d` so it should be fine for now.
Seaborn doesn't seem to filter nan on line however.
Can't move it later in the chain as this nan-trick is used to split on different dimensions, can move it to the if check above though

NaN's are usually usually not wanted in line plots either in my experience. Do you have a usecase in mind?

Reply:
Moved the null filter inside a scatter check.

Reply:
> NaN's are usually not wanted in line plots either in my experience. Do you have a usecase in mind?

Plenty. it's important to know where the gaps are. That's useful information. It is annoying that matplotlib will skip NaNs at the beginning and ends of a line plot.